server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -87,9 +92,46 @@ public Map<String, NodeUsageStatsForThreadPools> simulatedNodeUsageStatsForThrea
                 adjustedNodeUsageStatsForThreadPools.put(entry.getKey(), entry.getValue());
             }
         }
+
```

Comment:
instead of adding a second non-trivial for-loop in `addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools`, could we replace the above original for-loop with iteration of the `simulatedNodeWriteLoadDeltas` data structure to begin with? 
Then use something like `originalNodeUsageStatsForThreadPools.forEach(adjustedNodeUsageStatsForThreadPools::putIfAbsent)` to straight copy the remainder.

This would be in combination with my other suggestion to add the new node to `originalNodeUsageStatsForThreadPools` beforehand for simplicity.